### PR TITLE
robottelo params and docs

### PR DESCRIPTION
**Pull request by pondrejk**
The aim is to make running robottelo tests easier for people not familiar with the test suite structure, selecting test type and endpoints should now suffice. Also added some docs

### CODE REVIEW

File: roles/robottelo/defaults/main.yml [hunk 1]
```diff
@@ -3,5 +3,27 @@ robottelo_url: https://github.com/SatelliteQE/robottelo.git
...
+  {% if robottelo_test_type == 'endtoend' %}
+    not stubbed
+  {% else %}
+    not stubbed and {{ robottelo_test_type }}
+  {% endif %}
+robottelo_test_path: >
+  {% if robottelo_test_type == 'destructive' %}
+    tests/foreman/sys
+  {% else %}
+    {% if robottelo_test_type == 'endtoend' %}
```
Comment (ekohl) [inline roles/robottelo/defaults/main.yml]: I thought there was a `elif` in Jinja templates which might be a bit more readable.

File: docs/testing.md [hunk 2]
```diff
@@ -88,3 +89,15 @@ If you want to spin up more than one client, let's say 10 for this example, the
...
+
+Robottelo is a test suite for excercising Foreman and Katello. Forklift provides a role for Robottelo set up and running tests against your machine. Configuration options of interest are `robottelo_test_endpoints` where you can pass a list of endpoints (api, cli or ui), and `robottelo_test_type`, which is one of:
+
+- tier1 to tier4 - base test sets, tier1 tests can resemble unit testing, higher tiers require more extensive setup
+- destructive - tests that restart or rename the server
+- upgrade - a selection of tests from tiers used in post-upgrade testing, should exercise the core functionality in less time consuming way
+- endtoend - testing the essential user scenario, less time-consuming than the upgrade set
+
+ * [Robottelo repository](https://github.com/SatelliteQE/robottelo)
+ * [Robottelo documentation](http://robottelo.readthedocs.io/en/latest/)
```
Comment (ekohl) [inline docs/testing.md]: I'm always a fan of https:// links when available.

File: docs/testing.md [hunk 3]
```diff
@@ -88,3 +89,15 @@ If you want to spin up more than one client, let's say 10 for this example, the
 ```
 docker-compose scale el6=10
 ```
+
+## Running Robottelo Tests
+
+Robottelo is a test suite for excercising Foreman and Katello. Forklift provides a role for Robottelo set up and running tests against your machine. Configuration options of interest are `robottelo_test_endpoints` where you can pass a list of endpoints (api, cli or ui), and `robottelo_test_type`, which is one of:
```
Comment (ehelms) [inline docs/testing.md]: "Robottelo to setup and run tests"

File: roles/robottelo/defaults/main.yml [hunk 4]
```diff
@@ -3,5 +3,25 @@ robottelo_url: https://github.com/SatelliteQE/robottelo.git
 robottelo_directory: /root/robottelo
 robottelo_virtualenv_path: /root/robottelo/venv
 robottelo_version: master
-robottelo_test_arguments:
-  - tests/foreman/endtoend/test_api_endtoend.py
+robottelo_junit_file: results.xml
+robottelo_test_type: endtoend # one of [tier1|tier2|tier3|destructive|upgrade|endtoend]
```
Comment (ehelms) [inline roles/robottelo/defaults/main.yml]: Can you only run one test type at a time? Or could this be an array of types?

**Comment by pondrejk** [comment]
@ekohl sounds good, about the passing set, I guess some more tweaking of robottelo.properties based on the machine will also be necessary to maximize the number. Also one thing coming to mind to avoid katello stuff, we could tweak which testfiles are collected (eg. name test_contentview.py suggests katello is required), this could be faster than some tagging on the robottelo side

File: roles/robottelo/defaults/main.yml [hunk 5]
```diff
@@ -3,5 +3,25 @@ robottelo_url: https://github.com/SatelliteQE/robottelo.git
...
+robottelo_test_endpoints: # [api|ui|cli]
+  - api
+robottelo_test_options: >
+  {% if robottelo_test_type == 'endtoend' %}
+    not stubbed
+  {% else %}
+    not stubbed and {{ robottelo_test_type }}
+  {% endif %}
+robottelo_test_path: >
```
Comment (ehelms) [inline roles/robottelo/defaults/main.yml]: A hash mapping these might be cleaner for readability.

File: roles/robottelo/defaults/main.yml (hunk 4 above)
Comment (pondrejk) [reply roles/robottelo/defaults/main.yml]: It is possible to run several test types at once, but I'm not sure I want to encourage that as I don't see any advantage in it, also it is not done in qa automation. I'd say it is better to run several times and have separate junit files per type.

**Comment by ekohl** [comment]
> @ekohl sounds good, about the passing set, I guess some more tweaking of robottelo.properties based on the machine will also be necessary to maximize the number. Also one thing coming to mind to avoid katello stuff, we could tweak which testfiles are collected (eg. name test_contentview.py suggests katello is required), this could be faster than some tagging on the robottelo side

I still think tagging on the robettelo side is more effective in the long run. This allows for easy manual runs which in my experience you stiill do.

The GitHub UI also suggests a rebase could be good.

**Comment by pondrejk** [comment]
rebased

File: roles/robottelo/defaults/main.yml [hunk 6]
```diff
@@ -3,5 +3,26 @@ robottelo_url: https://github.com/SatelliteQE/robottelo.git
...
 robottelo_virtualenv_path: /root/robottelo/venv
 robottelo_version: master
-robottelo_test_arguments:
-  - tests/foreman/endtoend/test_api_endtoend.py
+robottelo_junit_file: results.xml
+robottelo_test_type: endtoend # one of [tier1|tier2|tier3|destructive|upgrade|endtoend]
+robottelo_test_endpoints: # [api|ui|cli]
+  - api
+robottelo_test_args: 
+  options: >
```
Comment (ekohl) [inline roles/robottelo/defaults/main.yml]: We usually create separate variables for cases like this.

Comment (pondrejk) [reply roles/robottelo/defaults/main.yml]: Yes, I had it separated, I changed it in reaction to @ehelms's comment about hash mapping, though I might misunderstood what he asked for

Comment (ehelms) [reply roles/robottelo/defaults/main.yml]: I was thinking more like:

```
robottelo_test_options:
  endtoend: 'not stubbed'
  default: "not stubbed and {{ robottelo_tyest_type }}"
```

**Comment by pondrejk** [comment]
@ekohl, @ehelms, I updated the pr, is there anything else blocking?

File: roles/robottelo/defaults/main.yml [hunk 7]
```diff
@@ -3,5 +3,20 @@ robottelo_url: https://github.com/SatelliteQE/robottelo.git
...
+robottelo_test_type: endtoend # one of [tier1|tier2|tier3|destructive|upgrade|endtoend]
+robottelo_test_endpoints: # [api|ui|cli]
+  - api
+robottelo_test_options:
+  endtoend: not stubbed
+  default: not stubbed and {{ robottelo_test_type }}
+robottelo_test_path:
+  destructive: tests/foreman/sys
+  endtoend: >
+    {% for item in robottelo_test_endpoints %}
```
Comment (ekohl) [inline roles/robottelo/defaults/main.yml]: I'm not sure how defaults here work with regards to variables in ansible. I haven't run robottelo to know if there should be a better way but this still feels complex.

**Comment by ehelms** [comment]
@pondrejk Can you please rebase?

**Comment by pondrejk** [comment]
rebased

**Comment by ehelms** [comment]
Thanks @pondrejk !

**Pull request merged**